src/main/java/controllers/UserProfileController.java

```diff
@@ -87,128 +88,164 @@ else if (choice == 1) {
-        }
-        else {
+        } else {
             this.inOut.setOutput("Username entered is the same as the old one. Returning to main page.");
         }
     }
 
     /**
-     * Displays the prompt to change password
+     * Run the change password logic
      */
     private void runChangePasswordDisplay(User user, String newPassword) {
         if (this.userManager.changePassword(user, newPassword)) {
             this.userManager.changePassword(user, newPassword);
             this.inOut.setOutput("Successfully changed password to: " + newPassword + "\n");
             this.inOut.setOutput("Returning to main page.");
-        }
-        else {
+        } else {
             this.inOut.setOutput("Password entered is the same as the old one. Returning to main page");
         }
     }
 
     /**
-     * Displays the prompt to change bio
+     * Run the change bio logic
      */
     private void runChangeBioDisplay(User user, String newBio) {
         if (this.userManager.changeBio(user, newBio)) {
             user.setBio(newBio);
             this.databaseManager.updateUser(user);
             this.inOut.setOutput("Successfully changed bio to: " + newBio + "\n");
             this.inOut.setOutput("Returning to main page.");
-        }
-        else {
+        } else {
             this.inOut.setOutput("Bio given is the same as the old one. Returning to main page");
         }
     }
 
     /**
-     * Return error msg if target_user is already followed by user, return successful msg
+     * Runs the follow user logic: display error msg if target_user is already followed by user, return successful msg
      * otherwise and follow target_user
+     *
+     * @param user the currently logged-in user
+     * @param targetUser the user that the currently logged-in user wishes to follow
      */
     private void runFollowUser(User user, User targetUser) {
         if (this.userManager.followUser(user, targetUser)) {
             this.userManager.followUser(user, targetUser);
             this.inOut.setOutput("Successfully followed the target user!");
-        }
-        else {
+        } else {
             this.inOut.setOutput("Target user already followed!");
         }
     }
 
     /**
-     * Return false if target_user is not followed by user, return true otherwise and unfollow target_user
+     * Runs the unfollow user logic
+     */
+    private void runUnfollowUser() {
+        try {
+            String usernameToUnfollow =
+                    this.inOut.getInput("Enter the username of the person you'd like to unfollow: ");
+            User userToUnfollow = this.userManager.findUser(usernameToUnfollow);
+            if (this.userManager.unfollowUser(this.loginManager.getCurrUser(), userToUnfollow)) {
+                this.inOut.setOutput("Successfully unfollowed the target user!");
+            } else {
+                this.inOut.setOutput("User is not followed to begin with!");
```

Comment:
what if they make a typo in the username?

Reply:
It will output that the user is not followed to begin with, do you think I should add something like: "User is not followed to begin with! (Check for typo) ?

Reply:
yeah that would be nice!